src/Talis/Persona/Client/Base.php

```diff
@@ -92,4 +102,54 @@ protected function checkConfig($config){
             throw new \InvalidArgumentException("Config provided does not contain values for: " . implode(",", $missingProperties));
         }
     }
+
+    /**
+     * @return Logger|\Psr\Log\LoggerInterface
+     */
+    protected function getLogger()
+    {
+        if ($this->logger==null)
+        {
+            $this->logger = new Logger(self::LOGGER_NAME);
+        }
+        return $this->logger;
+    }
+
+    /**
+     * Perform the request according to the $curlOptions
+     * @param $curlOptions
+     * @param $expectResponse set true if you expect a JSON response with a 200, otherwise expect a 204 no content
+     * @return array
+     * @throws \Exception if response not 200 and valid JSON
+     */
+    protected function performRequest($curlOptions,$expectResponse=true)
+    {
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        $expectedResponseCode = ($expectResponse) ? 200 : 204;
```

Comment:
Is this something that we should be sending from the methods that call this? Are there other response codes that might be successful - just a thought but does the getuser call when you pass a gupid return a 303? Or does it resolve to a 200?